### PR TITLE
improve progress bar and hexagon

### PR DESCRIPTION
**Pull request by juliangruber**
cc @Kriesse

### CODE REVIEW

File: pages/main.js [hunk 1]
```diff
@@ -55,11 +55,20 @@ function createTable (dats, send) {
...
     return html`
       <tr class="dat-list__item">
         <td>
           <div class="dat-hexagon">
-            ${dat.owner ? '↑' : '↓'}
+            ${{
+              loading: '↓',
+              paused: '',
+              complete: '↑'
+            }[state]}
```
Comment (yoshuawuyts) [inline pages/main.js]: I find this hard to read

Comment (juliangruber) [inline pages/main.js]: it's something I learned from @substack years ago, once you get used to it it's a lot easier to read and type than say

```js
${
  state === 'loading' && 'v' ||
  state === 'paused' && '' ||
  state === 'complete' && '^'
}
```

it's basically a switch statement as a value, which JavaScript is missing.

Comment (karissa) [inline pages/main.js]: Works for me

**Comment by yoshuawuyts** [comment]
hah yeah I get what you're doing and it's clever - but clever code is also
hard to debug. What I'd propose is that instead of selecting the value
inline you'd define it as a variable somewhere higher in the function and
use that

**Comment by clkao** [comment]
hey @juliangruber the dependency-ci complains that `hyperdrive-stats` is not licensed

**Comment by juliangruber** [comment]
> hey @juliangruber the dependency-ci complains that hyperdrive-stats is not licensed

thanks, fixed!

**Comment by juliangruber** [comment]
> hah yeah I get what you're doing and it's clever - but clever code is also
> hard to debug. What I'd propose is that instead of selecting the value
> inline you'd define it as a variable somewhere higher in the function and
> use that

@yoshuawuyts you mean like this?

```js
const hexContent = {
  loading: '↓',
  paused: '',
  complete: '↑'
}[state]
```

**Comment by yoshuawuyts** [comment]
Yeah, doing that inline is hard to read. Meant to reply to the code snippet
but apparently email doesn't let me hah

**Comment by juliangruber** [comment]
cool, check this out @yoshuawuyts

**Comment by karissa** [comment]
Yeah +1, I'm down with pulling all js out of the templates if possible, so the html can focus on being html while variables are defined above. Thanks for pointing that out, @yoshuawuyts !

**Comment by clkao** [comment]
LGTM. The progress is not actually updating (numbers in stats object stays the same), but this is probably another issue.

**Comment by juliangruber** [comment]
@clkao it is, just taking a long time. I haven't gotten behind that yet.

**Pull request merged**